core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -405,6 +408,20 @@ public void setRequestTimeoutHandler(Runnable requestTimeoutHandler) {
         this.requestTimeoutHandler = requireNonNull(requestTimeoutHandler, "requestTimeoutHandler");
     }
 
+    @Override
+    public boolean isRequestTimedOut() {
+        return timedOut;
+    }
+
+    /**
+     * Marks this {@link ServiceRequestContext} as having been timed out. Any callbacks created with
+     * {@code makeContextAware} that are run after this will be failed with {@link CancellationException}.
+     */
+    public void setRequestTimedOut() {
```

Comment:
You don't really need to introduce new methods in this case. If we just overrode `setTimedOut()` and `isTimedOut()` without `@Deprecated` tag, `ServiceRequestContext.isTimedOut()` and `setTimedOut()` will not emit a deprecation warning:

```java
interface RequestContext {
    /**
     * Old Javadoc stays here.
     * @deprecated Use {@link ServiceRequestContext#isRequestTimedOut()}.
     */
    @Deprecated
    boolean isTimedOut();
}

interface ServiceRequestContext extends RequestContext {
    /**
     * Put your new Javadoc here.
     */
    @Override
    boolean isTimedOut();
}

ServiceRequestContext ctx = ...;
boolean timedOut1 = ctx.isTimedOut(); // No warning
boolean timeoutOut2 = ((RequestContext) ctx).isTimedOut(); // Warning
```

Reply:
Changed it as you suggested. Thanks!